### PR TITLE
Merge elemental config into agent config

### PR DESCRIPTION
**Pull request by Itxaka**
- Just one place for everything
 - Dropped the extra module for elemental config
 - Some shared utils moved into their own modules to avoid circular
   dependencies
 - logger now available everywhere!
 - Config has the whole thing on it, less mangling of the config
 - spec now in the config module

builds upon #100 and #101 so those need to go in before this one

Signed-off-by: Itxaka <itxaka.garcia@spectrocloud.com>

### CODE REVIEW

File: internal/agent/hooks/hook.go [hunk 1]
```diff
@@ -10,7 +10,6 @@ type Interface interface {
 }
 
 var AfterInstall = []Interface{
-	&RunStage{},    // Shells out to stages defined from the container image
```
Comment (Itxaka) [inline internal/agent/hooks/hook.go:13]: This was removed and run directly on the action like the others.

File: internal/agent/install.go [hunk 2]
```diff
@@ -76,6 +66,8 @@ func ManualInstall(c, device string, reboot, poweroff, strictValidations bool) e
 		return err
 	}
 
+	fmt.Print(litter.Sdump(cc))
```
Comment (mauromorales) [inline internal/agent/install.go:69]: is this intended?

File: main.go [hunk 3]
```diff
@@ -527,11 +526,11 @@ The validate command expects a configuration file as its only argument. Local fi
 		},
 		Action: func(c *cli.Context) error {
 			stage := c.Args().First()
-			config, err := config.Scan(collector.Directories(configScanDir...), collector.NoLogs)
+			config, err := agentConfig.Scan(collector.Directories(configScanDir...), collector.NoLogs)
 			if err != nil {
 				return err
 			}
-			cfg, err := elementalConfig.ReadConfigRunFromAgentConfig(config)
+			cfg, err := agentConfig.ReadConfigRunFromAgentConfig(config)
```
Comment (mauromorales) [inline main.go:533]: this is returning a `v1.Config`, why do we still need that distinction vs `config` before? ... and if we have to keep it maybe it helps to name them something different to make this obvious

File: pkg/config/config.go [hunk 4]
```diff
@@ -36,6 +42,59 @@ type Install struct {
 	BindMounts             []string          `yaml:"bind_mounts,omitempty"`
 }
 
+func NewConfig(opts ...GenericOptions) *Config {
+	log := v1.NewLogger()
+
+	defaultPlatform, err := v1.NewPlatformFromArch(runtime.GOARCH)
```
Comment (mauromorales) [inline pkg/config/config.go:48]: Can we name this something other than default? maybe underlyingPlatform or hostPlatform, or something like that?

File: pkg/config/config.go [hunk 5]
```diff
@@ -46,6 +105,149 @@ type Config struct {
...
+		return err
+	}
+
+	return nil
+}
+
+// LoadInstallState loads the state.yaml file and unmarshals it to an InstallState object
+func (c Config) LoadInstallState() (*v1.InstallState, error) {
+	installState := &v1.InstallState{}
+	data, err := c.Fs.ReadFile(filepath.Join(constants.RunningStateDir, constants.InstallStateFile))
```
Comment (mauromorales) [inline pkg/config/config.go:156]: I see this was copied from before so no need to change in this PR but it feels like the full function could be a method of config

File: pkg/utils/grub.go [hunk 6]
```diff
@@ -18,21 +18,22 @@ package utils
 
 import (
 	"fmt"
+	agentConfig "github.com/kairos-io/kairos-agent/v2/pkg/config"
+	fs2 "github.com/kairos-io/kairos-agent/v2/pkg/utils/fs"
```
Comment (mauromorales) [inline pkg/utils/grub.go:22]: maybe fsutils would fit better than fs2

Comment (Itxaka) [reply pkg/utils/grub.go:22]: Good catch, wrong rename when moving the package

File: internal/agent/install.go (hunk 2 above)
Comment (Itxaka) [reply internal/agent/install.go:69]: Nope! Good catch

File: main.go (hunk 3 above)
Comment (Itxaka) [reply main.go:533]: umm, impossible, v1.Config does not exist anymore, there is now just one Config struct in the whole repo. Where did you see that?

File: pkg/config/config.go (hunk 4 above)
Comment (Itxaka) [reply pkg/config/config.go:48]: done!

File: pkg/config/config.go (hunk 5 above)
Comment (Itxaka) [reply pkg/config/config.go:156]: ? isnt it already a method of Config?

File: main.go (hunk 3 above)
Comment (mauromorales) [reply main.go:533]: if I understand correctly 

```
config, err := agentConfig.Scan(collector.Directories(configScanDir...), collector.NoLogs)
```

see https://github.com/kairos-io/kairos-agent/pull/102/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R529 

returns a `config.Config`

while `cfg, err := agentConfig.ReadConfigRunFromAgentConfig(config)`

returns a types `v1.Config` which I saw was already replaced for some of the other cases

File: pkg/config/config.go (hunk 5 above)
Comment (mauromorales) [reply pkg/config/config.go:156]: but not a big deal this one

File: main.go (hunk 3 above)
Comment (Itxaka) [reply main.go:533]: ohh I see. Both are returning a config. There is no longer a v1.Config, but this is being called twice for no reason probably!

Comment (Itxaka) [reply main.go:533]: and this returns the same config in both cases:

`func ReadConfigRunFromAgentConfig(c *Config) (*Config, error) `

Comment (Itxaka) [reply main.go:533]: in any case, sometimes the config needs to be initialized in a different config to fullfill some of the required stuff.

for example the `agentConfig.Scan` will return a config, yes, but that config wont have the logger on it for example.

I tihnk the fix here is to hijack the Scan to create the config with the `Newconfig` method, so we dont have an empty config.Logger for example.

Comment (Itxaka) [reply main.go]: good catch here, seems like I was doing this twice due to initializing the Config object on the scan directly. Now it does so all the config defaults are set from scratch and much more code can be removed, yay!

**Comment by Itxaka** [comment]
This is now broken lmao

File: main.go (hunk 3 above)
Comment (Itxaka) [reply main.go]: ok, fixed and even improved!

Comment (mauromorales) [reply main.go]: cool cool cool

**Review by mauromorales — APPROVED** [review]
lgtm

**Pull request merged**